Core/Core/View/Base/AppReview/AppReviewViewModel.swift

```diff
@@ -0,0 +1,160 @@
+                CoreLocalization.Review.thanksForVoteDescription
+            case .thanksForFeedback:
+                CoreLocalization.Review.thanksForFeedbackDescription
+            }
+        }
+    }
+    
+    @Published var state: ReviewState = .vote
+    @Published var rating: Int = 0
+    @Published var showReview: Bool = false
+    @Published var showSelectMailClientView: Bool = false
+    @Published var feedback: String = ""
+    @Published var clients: [ThirdPartyMailClient] = []
+    let allClients = ThirdPartyMailClient.clients
+    
+    private let config: Config
+    private var storage: CoreStorage
+    
+    public init(config: Config, storage: CoreStorage) {
+        self.config = config
+        self.storage = storage
+    }
+    
+    public func shouldShowRatingView() -> Bool {
+        guard let currentVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String else {
+            return false
+        }
+        guard let lastShownVersion = storage.reviewLastShownVersion else {
+            storage.reviewLastShownVersion = currentVersion
+            
+            if let lastReviewDate = storage.lastReviewDate {
+                return hasPassedFourMonths(from: lastReviewDate)
+            } else {
+                return true
+            }
+        }
+        return isNewerVersion(currentVersion: currentVersion, lastVersion: lastShownVersion)
+    }
+    
+    private func hasPassedFourMonths(from date: Date) -> Bool {
+        let currentDate = Date()
+        let calendar = Calendar.current
+         
+        if let futureDate = calendar.date(byAdding: .month, value: 4, to: date) {
+          return currentDate >= futureDate
+        }
+         
+        return false
+      }
+    
+    func reviewAction() {
+        withAnimation(Animation.easeIn(duration: 0.2)) {
+            if rating <= 3 {
+                state = .feedback
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+                    withAnimation(Animation.easeIn(duration: 0.1)) {
+                        self.showReview = true
+                    }
+                }
+            } else {
+                state = .thanksForVote
+            }
+        }
+    }
+    
+    func writeFeedbackToMail() {
+        self.clients = allClients.filter({ ThirdPartyMailer.isMailClientAvailable($0) })
+        if !clients.isEmpty {
+            withAnimation(Animation.bouncy(duration: 0.2)) {
+                showSelectMailClientView = true
+            }
+        } else {
+            openMailClient(ThirdPartyMailClient.systemDefault)
+        }
+    }
+    
+    func requestReview() {
+        SKStoreReviewController.requestReview()
+        storage.lastReviewDate = Date()
+    }
+    
+    func openMailClient(_ with: ThirdPartyMailClient) {
+        
+        let osVersion = UIDevice.current.systemVersion
+        let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
```

Comment:
`Bundle.main.infoDictionary?["CFBundleShortVersionString"]` is being used at multiple places, i think we should centralize this somewhere, what are your thoughts?

Reply:
We'll need to think about this as part of the next refactoring.